03_letters/joshcarp/letters_test.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var stdout = os.Stdout
+
+var lettersTests = []struct {
+	input       string
+	expectedMap map[rune]int
+	expectedStr []string
+}{
+	{"aba", map[rune]int{'a': 2, 'b': 1}, []string{"a:2", "b:1"}},
+	{"ywbgjzhoiumsnqdexprlctvfka", map[rune]int{
+		'a': 1, 'b': 1, 'c': 1, 'd': 1, 'e': 1, 'f': 1, 'g': 1, 'h': 1, 'i': 1,
+		'j': 1, 'k': 1, 'l': 1, 'm': 1, 'n': 1, 'o': 1, 'p': 1, 'q': 1, 'r': 1,
+		's': 1, 't': 1, 'u': 1, 'v': 1, 'w': 1, 'x': 1, 'y': 1, 'z': 1},
+		[]string{"a:1", "b:1", "c:1", "d:1", "e:1", "f:1", "g:1", "h:1", "i:1",
+			"j:1", "k:1", "l:1", "m:1", "n:1", "o:1", "p:1", "q:1", "r:1",
+			"s:1", "t:1", "u:1", "v:1", "w:1", "x:1", "y:1", "z:1"}},
+	{`.#jV#Rz;ZO`, map[rune]int{
```

Comment:
What about a couple more edge cases?

Empty input?
Non ASCII characters (😀😃) to test that a rune map is being used as the underlying datastore?